### PR TITLE
[cli] [vtctld] Migrate vtctld flags to pflags

### LINKED ISSUE #10893: Switch flag definitions to be on pflag instead of flag in `package go/vt/vtctld`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/vtctld/*.go
go/vt/vtctld/action_repository.go:      actionTimeout = flag.Duration("action_timeout", wrangler.DefaultActionTimeout, "time to wait for an action before resorting to force")
go/vt/vtctld/api.go:    localCell        = flag.String("cell", "", "cell to use")
go/vt/vtctld/api.go:    showTopologyCRUD = flag.Bool("vtctld_show_topology_crud", true, "Controls the display of the CRUD topology actions in the vtctld UI.")
go/vt/vtctld/api.go:    proxyTablets     = flag.Bool("proxy_tablets", false, "Setting this true will make vtctld proxy the tablet status instead of redirecting to them")
go/vt/vtctld/tablet_data.go:    tabletHealthKeepAlive = flag.Duration("tablet_health_keep_alive", 5*time.Minute, "close streaming tablet health connection if there are no requests for this long")
go/vt/vtctld/vtctld.go: enableRealtimeStats = flag.Bool("enable_realtime_stats", false, "Required for the Realtime Stats view. If set, vtctld will maintain a streaming RPC to each tablet (in all cells) to gather the realtime health stats.")
go/vt/vtctld/vtctld.go: enableUI            = flag.Bool("enable_vtctld_ui", true, "If true, the vtctld web interface will be enabled. Default is true.")
go/vt/vtctld/vtctld.go: _                   = flag.String("durability_policy", "none", "type of durability to enforce. Default is none. Other values are dictated by registered plugins")
go/vt/vtctld/vtctld.go: sanitizeLogMessages = flag.Bool("vtctld_sanitize_log_messages", false, "When true, vtctld sanitizes logging.")
go/vt/vtctld/vtctld.go: _ = flag.String("web_dir", "", "NOT USED, here for backward compatibility")
go/vt/vtctld/vtctld.go: _ = flag.String("web_dir2", "", "NOT USED, here for backward compatibility")
go/vt/vtctld/vtctld.go: deprecatedOnlineDDLCheckInterval = flag.Duration("online_ddl_check_interval", 0, "deprecated. Will be removed in next Vitess version")
go/vt/vtctld/vtctld_test.go:    flag.Set("enable_vtctld_ui", "false")
go/vt/vtctld/vtctld_test.go:    defer flag.Set("enable_vtctld_ui", "true")
go/vt/vtctld/workflow.go:       workflowManagerInit        = flag.Bool("workflow_manager_init", false, "Initialize the workflow manager in this vtctld instance.")
go/vt/vtctld/workflow.go:       workflowManagerUseElection = flag.Bool("workflow_manager_use_election", false, "if specified, will use a topology server-based master election to ensure only one workflow manager is active at a time.")
go/vt/vtctld/workflow.go:       flag.Var(&workflowManagerDisable, "workflow_manager_disable", "comma separated list of workflow types to disable")
```

### PR DESCRIPTION
**Pull request by rsajwani**
Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>

## Description

Updates vtctld related flag references specified in https://github.com/vitessio/vitess/issues/10893

Modules for flag registration

go list -f '{{ .ImportPath }}| {{ join .Deps " " }} ' ./go/cmd/... | awk -F"|" '$2 ~ "vitess.io/vitess/go/vt/vtctld" {print $1}'
vitess.io/vitess/go/cmd/vtcombo
vitess.io/vitess/go/cmd/vtctld

## Related Issue(s)

fixes #10893

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by ajm188** [comment]
I think your description is maybe pasted from some other PR?

> Updates throttler related flag references specified in https://github.com/vitessio/vitess/issues/10893

this is for `vtctld` no?

File: go/vt/vtctld/workflow.go [hunk 1]
```diff
@@ -33,17 +35,26 @@ import (
 )
 
 var (
-	workflowManagerInit        = flag.Bool("workflow_manager_init", false, "Initialize the workflow manager in this vtctld instance.")
-	workflowManagerUseElection = flag.Bool("workflow_manager_use_election", false, "if specified, will use a topology server-based master election to ensure only one workflow manager is active at a time.")
-	workflowManagerDisable     flagutil.StringListValue
+	workflowManagerInit        = false
+	workflowManagerUseElection = false
```
Comment (ajm188) [inline go/vt/vtctld/workflow.go:39]: same comment about zero values

File: go/vt/vtctld/workflow.go [hunk 2]
```diff
@@ -33,17 +35,26 @@ import (
...
+
+	workflowManagerDisable flagutil.StringListValue
```
Comment (ajm188) [inline go/vt/vtctld/workflow.go:41]: can you try switching this to `[]string` and using pflag's `StringSliceVar` ?

File: go/vt/vtctld/workflow.go [hunk 3]
```diff
@@ -33,17 +35,26 @@ import (
...
+func registerVtctldWorkflowFlags(fs *pflag.FlagSet) {
+	fs.BoolVar(&workflowManagerInit, "workflow_manager_init", workflowManagerInit, "Initialize the workflow manager in this vtctld instance.")
+	fs.BoolVar(&workflowManagerUseElection, "workflow_manager_use_election", workflowManagerUseElection, "if specified, will use a topology server-based master election to ensure only one workflow manager is active at a time.")
+}
+
 func init() {
+	for _, cmd := range []string{"vtcombo", "vtctld"} {
+		servenv.OnParseFor(cmd, registerVtctldWorkflowFlags)
+	}
 	flag.Var(&workflowManagerDisable, "workflow_manager_disable", "comma separated list of workflow types to disable")
```
Comment (ajm188) [inline go/vt/vtctld/workflow.go:53]: ?

**Review by rohit-nayak-ps — APPROVED** [review]
lgtm

**Pull request merged**